### ISSUE TITLE
N64 stuttering at start of games on CrossMix-OS v1.2.1

### ISSUE THREAD

**Issue by Soupnazi1981** [cizia64/CrossMix-OS#105]
On a fresh install of CrossMix-OS v1.2.1 with default settings I notice there is stuttering when N64 games are first launched. I don't notice this on stock os 1.0.4.

Eg. Launch Mario Kart 64, select grand prix race, Mushroom cup, 50 CC. When the race first starts there is stuttering for first 3-4 seconds. Then it seems to catchup and then stuttering goes away.

Another example is in Mario 64. Start game, watch cutscene with Peach, the jump into the first level. (Bomb Omb level). Once the level starts, it stutters for 3-4 seconds. I don't have this stuttering on stock.

**Comment by wankio** [comment]
same as nds, performance is suck, slow, always run at 60% instead 100%

**Comment by Soupnazi1981** [comment]
> same as nds, performance is suck, slow, always run at 60% instead 100%

Yeah, I just opened up a new request for NDS performance. NDS is unplayable for some reason.

**Comment by cizia64** [comment]
Probably same as #106  :  

Yes it has been fixed. Only those who have downloaded the v1.2.1 in the first 7 hours are concerned by this problem.
This is due to alsa configuration. 

**To fix it just run OTA update from apps.**

For those who don't want to enable wifi on their TSP,  here the manual way : copy [this asound.conf file](https://github.com/cizia64/CrossMix-OS/blob/main/System/usr/trimui/etc/asound.conf) in `/etc` on the internal storage (ROOT) , reboot and try again NDS


Let me know if it fixes your N64 problem too.

**Comment by Soupnazi1981** [comment]
> Probably same as #106 :
> 
> Yes it has been fixed. Only those who have downloaded the v1.2.1 in the first 7 hours are concerned by this problem. This is due to alsa configuration.
> 
> **To fix it just run OTA update from apps.**
> 
> For those who don't want to enable wifi on their TSP, here the manual way : copy [this asound.conf file](https://github.com/cizia64/CrossMix-OS/blob/main/System/usr/trimui/etc/asound.conf) in `/etc` on the internal storage (ROOT) , reboot and try again NDS
> 
> Let me know if it fixes your N64 problem too.

Thanks for the info. NDS performance is fixed with the OTA update.

Unfortunately N64 performance with the default Parallelei N64 emulator still stutters at the beginning of games. I tested the same two games as before. It's quite easy to reproduce.

1. Start Super Mario 64, create new game, finish cutscene with Peach. Then go to first BombOmb stage. As soon as stage starts the FPS goes from 30 to 22-24 for the first 10 seconds or so. Sound stutters and frames are choppy at this time. Then after 10 seconds it normalizes at 29-30fps and is playable. If you quit the Bomb Omb stage and go back in, it does the same behavior.

2. Start Mario Kart, choose Mushroom Cup, 50CC with any character. When the race first starts at the starting grid, the performance is 22-24FPS and stutters until about 5 seconds and then it goes up to 29-30 fps. If you restart the race and retry, it does the same behavior.

It seems the default Parallei N64 emulator has slightly poorer performance than stock. I'm not sure if this is the same default emulator as the stock one. I know you can manually select the various emulators with the X button but it would be nice to have an option to select which is the default emulator when you just launch a game.

**Comment by cizia64** [comment]
Indeed the core is configured in "ondemand" performance and should be set on "performance" mode.
Can you set Fn key ON and try again to tell me if it is better ?

**Comment by Soupnazi1981** [comment]
> Indeed the core is configured in "ondemand" performance and should be set on "performance" mode. Can you set Fn key ON and try again to tell me if it is better ?

Oh I actually always use performance made with FN key enabled which I used during the test
